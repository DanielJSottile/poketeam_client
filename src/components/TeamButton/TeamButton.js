import React, { Component } from 'react';
import {Link} from 'react-router-dom';
import UserContext from '../../contexts/UserContext';

export default class TeamButton extends Component {

  static contextType = UserContext;

  render() {

    const {handleCurrentTeamClicked} = this.context;
    
    // For now, the Link doesnt go to the anchor because the hash looks disgusting.
    return ( 
      <a 
        href={`#${this.props.team_name}`}
        className="btn"
        id={this.props.id} 
        onClick={(e) => {
          e.preventDefault() 
          handleCurrentTeamClicked(this.props.team_name, this.props.id);
        }}>
        <i className="fas fa-layer-group"></i> {this.props.team_name}
      </a>
    );
  };
};
