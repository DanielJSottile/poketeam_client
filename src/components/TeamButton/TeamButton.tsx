import React, { useContext } from "react";
import GeneralContext from "../../contexts/GeneralContext";
import "./TeamButton.css";

const TeamButton = (props: any) => {
  const GenCon = useContext(GeneralContext);

  const { handleCurrentTeamClicked } = GenCon;

  // For now, the Link doesnt go to the anchor because the hash looks disgusting.
  return (
    <a
      href={`#${props.team_name}`}
      className="btn"
      id={props.id}
      onClick={(e) => {
        e.preventDefault();
        handleCurrentTeamClicked(props.team_name, props.id);
      }}
    >
      <i className="fas fa-layer-group"></i> {props.team_name}
    </a>
  );
};

export default TeamButton;
