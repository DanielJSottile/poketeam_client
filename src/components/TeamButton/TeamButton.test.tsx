import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
//import { render } from '@testing-library/react';
import TeamButton from "./TeamButton";
import { shallow, mount, render } from "enzyme";
import toJson from "enzyme-to-json";

describe("<TeamButton/>", (): void => {
  it("renders without crashing", (): void => {
    const div: HTMLDivElement = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <TeamButton />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  it("snapshot", (): void => {
    const wrapper = shallow(<TeamButton />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
