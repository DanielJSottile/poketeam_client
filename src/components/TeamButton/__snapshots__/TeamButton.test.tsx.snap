// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TeamButton/> snapshot 1`] = `
<a
  className="btn"
  href="#undefined"
  onClick={[Function]}
>
  <i
    className="fas fa-layer-group"
  />
   
</a>
`;
